src/app/search/search.component.ts

```diff
@@ -0,0 +1,121 @@
+import { Component, OnInit } from '@angular/core';
+import { CmsService, SpinnerComponent, HalDoc, StoryModel, SeriesModel } from '../shared';
+import { SearchStoryComponent } from './directives/search-story.component';
+
+@Component({
+  directives: [SpinnerComponent, SearchStoryComponent],
+  selector: 'publish-search',
+  styleUrls: ['search.component.css'],
+  templateUrl: 'search.component.html'
+})
+
+export class SearchComponent implements OnInit {
+
+  isLoaded = false;
+  totalCount: number;
+  pages: number[];
+  noStories: boolean;
+  auth: HalDoc;
+  stories: StoryModel[];
+  storyLoaders: boolean[];
+
+  currentPage: number;
+  showNumPages: number = 10;
+  pagesBegin: number;
+  pagesEnd: number;
+
+  searchTerm: string;
+  searchSeriesId: number;
+  searchSeries: SeriesModel;
+  allSeriesIds: number[];
+  allSeries: any;
+
+  constructor(private cms: CmsService) {}
+
+  ngOnInit() {
+    this.cms.follow('prx:authorization').subscribe((auth) => {
+      this.auth = auth;
+
+      this.allSeriesIds = [-1];
+      this.allSeries = {};
+      auth.followItems('prx:series', {filters: 'v4'}).subscribe((series) => {
+        for (let s of <HalDoc[]> series) {
+          this.allSeriesIds.push(s.id);
+          this.allSeries[s.id] = new SeriesModel(auth, s, false);
+        }
+
+        this.searchTerm = '';
+        this.searchSeries = null;
+        this.currentPage = 1;
+        this.search(1);
+      });
+    });
+  }
+
+  searchBySeries() {
+    this.searchSeries = this.allSeries[this.searchSeriesId];
+    this.search(1);
+  }
+
+  search(page: number, per: number = 10) {
+    this.currentPage = page;
+    this.isLoaded = false;
+    this.noStories = false;
+
+    let parent = this.searchSeries ? this.searchSeries.doc : this.auth;
+
+    let draftStory = new StoryModel(this.auth);
+    let isUnsavedDraft = draftStory.isNew && draftStory.isStored() && this.searchSeries === null;
+    let showUnsavedDraft = isUnsavedDraft && this.currentPage === 1 && this.searchSeries === null;
```

Comment:
Cool. Drafts do make this more complicated than it needs to be. I was going off the requirement Has-unsaved-changes indicator (localstorage)